coalib/bearlib/languages/Language.py

```diff
@@ -422,8 +422,11 @@ def __or__(self, other):
 
     def __contains__(self, item):
         item = Language[item]
+        itemVers = set(item.versions)
+        versions = set(self.versions)
         return (type(self) is type(item)
-                and set(item.versions).issubset(set(self.versions)))
+                and (itemVers.issubset(versions)
```

Comment:
The logic seems very strange:

x subset y or x superset y

what is the better way of saying that?

is that correct?

Reply:
I think the logic is sound. 

I could also write it as ```x subset y or y subset x``` if it makes it clearer. Either all of the versions in the passed in item will be contained in self (e.g. 'python 3' in Language['python']) or all the versions in self will be contained in item (e.g. 'python' in Language['python 3']) 

if neither of these conditions are met then item isn't in self.

I can't think of a nicer way to write this. Maybe

```python
x | y == x or x | y == y
```